mailpoet/assets/js/src/newsletters/send/filter_segment.tsx

```diff
@@ -0,0 +1,147 @@
+import { NewsLetter } from 'common/newsletter';
+import { Field } from 'form/types';
+import { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import { Selection } from 'form/fields/selection';
+import { Toggle } from 'common';
+import { Tooltip } from 'common/tooltip/tooltip';
+import { Icon, help } from '@wordpress/icons';
+
+type FilterSegmentProps = {
+  item?: NewsLetter;
+  onValueChange: (targetWrap: {
+    target: {
+      name: string;
+      value: unknown;
+    };
+  }) => void;
+  field: Field;
+};
+
+export function FilterSegment({
+  item,
+  onValueChange,
+  field,
+}: FilterSegmentProps) {
+  const currentFilterSegmentId = item?.options.filterSegmentId;
+  const premiumFeaturesAvailable: boolean = useMemo(
+    () =>
+      window.mailpoet_premium_active && window.mailpoet_has_valid_premium_key,
```

Comment:
Do we also need to check here if the site reached the subscribers limit?

Reply:
It would be nice to have a constant or a help function that we could use to check that. We do have a few attempts at that that might need consolidation:

https://github.com/mailpoet/mailpoet-premium/blob/bdb626b23e37940251b2e084cb5c65f37c5d46ac/assets/js/src/premium.ts#L6
https://github.com/mailpoet/mailpoet/blob/f8c7adf5a6cc697c4b84307bb4a2cc8e269906cc/mailpoet/assets/js/src/common/premium_modal/index.tsx#L29
https://github.com/mailpoet/mailpoet/blob/9930f2915c5fca19a4d03433f964432f2c6b655c/mailpoet/assets/js/src/common/premium_modal/upgrade_info.ts#L21

I believe there is even one more that I tried to use when creating https://github.com/mailpoet/mailpoet/pull/5174, but it didn't work straightaway and I didn't want to spend more time on that PR.

I don't think you necessarily need to use a helper function or constant here since this seems to be a bigger problem. But mentioning it in case you want to give it a go.

Reply:
I don't recall the details, but I believe it is recommended that we use the MailPoet object instead of window to check if those flags are true or false.

Reply:
Thanks for catching this! I'm going to use the existing `premiumValidAndActive` https://github.com/mailpoet/mailpoet/blob/MAILPOET-5510%2Fupdate-sending-step-only/mailpoet/assets/js/src/common/premium_modal/index.tsx#L28-L28, which I think captures what we want.

Part of what I was doing here is I needed this to match what we have in `send.tsx` (get premiumFeaturesAvailable), so I updated that as well.

This change is in commit 7a53faf48707016a2282d5e8383c5099f23129e9